converter/util.go

```diff
@@ -21,10 +25,25 @@ func insert(out output, path []string, key string, value interface{}) {
 	out[key] = value
 }
 
-func convertModuleDiff(out output, diff *terraform.ModuleDiff) {
-	insert(out, diff.Path, "destroy", diff.Destroy)
+func convertModuleDiff(out output, diff *terraform.ModuleDiff, flat bool) {
 	for k, v := range diff.Resources {
-		convertInstanceDiff(out, append(diff.Path, k), v)
+		var instanceName []string
+		if flat {
+			var flatName strings.Builder
+
+			if len(diff.Path) > 1 {
+				// slice [1:] to omit "Root" string
+				flatName.WriteString(strings.Join(diff.Path[1:], "."))
+				flatName.WriteString(".")
+			}
+
+			flatName.WriteString(k)
+			instanceName = []string{flatName.String()}
+		} else {
+			insert(out, diff.Path, "destroy", diff.Destroy)
```

Comment:
@morgante in the flattened form, each of the instances has the `"destroy": false"` pair, so I omitted it from the root level as it seemed redundant. If that breaks compatibility with consuming tools, I'll add.

Reply:
@domZippilli What do you mean by that? Having info on whether an individual resource is being destroyed or not would be very critical for certain plan analysis.

Reply:
Sure. It's still transcribed for each instance. This is just about the root of the flat tree. Shouldn't be hard to add back.

Reply:
I added it to the flattened tree. It just takes a little logic so you don't end up with objects for each module that just has a destroy pair and nothing else.